internal/idp/providers/oidc/oidc.go

```diff
@@ -0,0 +1,109 @@
+package oidc
+
+import (
+	"context"
+
+	"github.com/zitadel/oidc/v2/pkg/client/rp"
+	"github.com/zitadel/oidc/v2/pkg/oidc"
+
+	"github.com/zitadel/zitadel/internal/idp"
+)
+
+var _ idp.Provider = (*Provider)(nil)
+
+// Provider is the idp.Provider implementation for a generic OIDC provider
```

Comment:
Please have a look at the doc links section of https://tip.golang.org/doc/comment
```suggestion
// Provider is the [idp.Provider] implementation for a generic OIDC provider
```